pkg/trustedresources/verify.go

```diff
@@ -45,25 +47,22 @@ const (
 	keyReference = "k8s://"
 )
 
-// VerifyInterface get the checksum of json marshalled object and verify it.
-func VerifyInterface(obj interface{}, verifier signature.Verifier, signature []byte) error {
-	ts, err := json.Marshal(obj)
-	if err != nil {
-		return err
-	}
+var (
+	// DefaultSignatureAlgorithm is the default algorithm for signature
+	DefaultSignatureAlgorithm = "sha256"
 
-	h := sha256.New()
-	h.Write(ts)
-
-	if err := verifier.VerifySignature(bytes.NewReader(signature), bytes.NewReader(h.Sum(nil))); err != nil {
-		return err
+	// supportedSignatureAlgorithms sets a list of support signature algorithms that is similar to the list supported by cosign
+	supportedSignatureAlgorithms = map[string]crypto.Hash{
+		"sha224": crypto.SHA224,
+		"sha256": crypto.SHA256,
+		"sha384": crypto.SHA384,
+		"sha512": crypto.SHA512,
 	}
+)
 
-	return nil
-}
-
-// VerifyTask verifies the signature and public key against task
-func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.Interface) error {
+// VerifyTask verifies the signature and public key against task, source is from ConfigSource.URI, which will be used to match policy patterns
+// TODO(#5527): refactor to avoid passing verificationPolicies down the stack.
+func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.Interface, source string, verificationPolicies []*v1alpha1.VerificationPolicy) error {
```

Comment:
The verify logics LGTM. Couldn't find a better way in the current context for doing this.